crates/core/c8y_api/src/smartrest/alarm.rs

```diff
@@ -1,168 +1,136 @@
-use tedge_api::alarm::AlarmSeverity;
-use tedge_api::alarm::ThinEdgeAlarm;
+use crate::json_c8y::AlarmSeverity;
+use crate::json_c8y::C8yCreateAlarm;
 use time::format_description::well_known::Rfc3339;
-use time::OffsetDateTime;
 
-/// Converts from thin-edge alarm to C8Y alarm SmartREST message
-pub fn serialize_alarm(alarm: ThinEdgeAlarm) -> Result<String, time::error::Format> {
-    match alarm.data {
-        None => Ok(format!("306,{}", alarm.name)),
+/// Serialize C8yCreateAlarm to SmartREST message
+pub fn serialize_alarm(alarm: C8yCreateAlarm) -> Result<String, time::error::Format> {
```

Comment:
Creating the SmartREST alarm directly from the `ThinEdgeAlarm` was intentional in the old regime, to avoid the creation of intermediate structs. That's why necessary defaults were also applied at this level itself, when the input didn't have some. But, now that you've made this change and the plan is to migrate JSON over MQTT anyway, which would need this `C8yCreateAlarm` struct anyway, let it be.